transform/staticlark/environment.go

```diff
@@ -0,0 +1,109 @@
+package staticlark
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"go.starlark.net/syntax"
+)
+
+const sensitiveVarName = "$"
+
+// environment is a collection of variable bindings, and the
+// list of other bindings that it derives value from
+// TODO(dustmop): Handle lexical scoping
```

Comment:
what's the plan for scoping? make the environment struct recursive, or a flat struct that applies scopes to string keys?

Reply:
Yes, add a stack to the environment, pushing and pulling as scopes are entered and left. Each binding needs a unique slot, and the name corresponds to that slot based upon which scope it is looked up in. Same type of structure used in a compiler or interpreter.